### PR TITLE
check for typos in ci

### PR DESCRIPTION
**Pull request by RayBB**
Per https://github.com/LemmyNet/lemmy-docs/pull/191#issuecomment-1582497270

Using: https://github.com/crate-ci/typos

You can see this works by looking at [this](https://woodpecker.join-lemmy.org/LemmyNet/lemmy-docs/pipeline/89/1) run where there is an intentional typo.

I wanted to install the latest version of typos but there is a weird problem that even when I set the docker image to the latest version of rust it was still using an old version that didn't work for typos.

@Nutomic recommended using a prebuilt binary for now so that's what I did.

An alternative would be to use https://hub.docker.com/r/imunew/typos-cli but who knows if it will be maintained. 

Open to suggestions for improving this.

### CODE REVIEW

File: .woodpecker.yml [hunk 1]
```diff
@@ -12,11 +12,18 @@ pipeline:
...
   check_formatting:
     image: tmknom/prettier
     commands:
       - prettier -c src
 
+  check_typos:
+    image: rust:1.70-slim-buster
+    commands:
+      - cargo install typos-cli@1.14.3
+      - typos src/en
```
Comment (Nutomic) [inline .woodpecker.yml:25]: Would be much faster to download a prebuilt binary.

https://github.com/crate-ci/typos/releases

Comment (RayBB) [reply .woodpecker.yml]: I took your suggestion and am using a prebuilt binary. Let me know further thoughts.

I'd recommend using squash and merge on github but I can manually squash if you don't want to do that.

**Comment by Nutomic** [comment]
Its not catching any typos which is a bit suspicious. Were they all fixed before, or is it possible that the check isnt running correctly?

**Comment by RayBB** [comment]
@Nutomic they were all fixed before.
Also this is because the project only fixes typos by a manually curated list of known typos rather than guessing what might be a typo. See https://github.com/crate-ci/typos#faq

**Review by Nutomic — APPROVED** [review]

**Comment by Nutomic** [comment]
Alright sounds good.

**Pull request merged**